src/server.c

```diff
@@ -5735,6 +5741,9 @@ sds genRedisInfoString(dict *section_dict, int all_sections, int everything) {
             "mem_cluster_links:%zu\r\n", mh->cluster_links,
             "mem_aof_buffer:%zu\r\n", mh->aof_buffer,
             "mem_allocator:%s\r\n", ZMALLOC_LIB,
+            "mem_db_hashtable_overhead_total:%zu\r\n", server.db_hashtable_overhead_total * sizeof(dictEntry*),
```

Comment:
I can understand that the term "hashtable" refers to the array of dictEntry pointers stored in the dict, and its overhead refers to the size of this pointer array.

However, it's a bit unfortunate that the memory stats also include an item with "hashtable" and "overhead" in its name: `overhead.hashtable.main`. This item not only calculates the size of the pointer array but also includes the size of all `dictEntry`, `robj`, and metadata.

Do we need to consider renaming it to differentiate it more clearly? For example, `mem_db_dict_ht_total`, but it might seem too low-level.

Reply:
that's unfortunate, but i don't like to rename existing metrics. and at the same time i don't like to mention `ht0` and `ht1`.
so let's keep `overhead.hashtable.main`, `overhead.hashtable.expiers`, and maybe call the new ones `overhead.hashtable.lut` and `overhead.hashtable.lut.rehashing`.
i think we should add them to both `MEMORY STATS`, and `INFO MEMORY`

p.s. i'm willing to drop `overhead.hashtable.lut`, and only add `overhead.hashtable.rehashing`

Reply:
I wasn't suggesting renaming existing metrics `overhead.hashtable.main` or `overhead.hashtable.expires`. I mean we can give the newly added metric in this PR a more appropriate name to differentiate it from the existing metrics.

BTW, I'm thinking if we can modify the semantics of `overhead.hashtable.main` to only represent the size of the dictEntry pointer array. Then, we can add a new metric like `overhead.database.main` to represent the overall overhead size including the size of all dictEntry, robj, and metadata. WDYT?

Reply:
i think that in some sense, reducing the scope of the existing metric from counting a total to counting just a portion, and adding a new one for the total, would be a breaking change.
let's just keep it intact and add 2 new metrics (to both MEMORY STATS and INFO MEMORY).

we just need to choose a suitable name.
so it can be `overhead.hashtable.lut`, add `overhead.hashtable.rehashing`
or maybe someone can suggest something better?

Reply:
I'm fine with `overhead.hashtable.lut` and `overhead.hashtable.rehashing`, let's just go on with this.